src/utils/lbann_library.cpp

```diff
@@ -48,193 +48,188 @@ std::shared_ptr<thread_pool> construct_io_thread_pool(lbann_comm *comm) {
       " (Limited to # Unused Compute Cores or 1)" << std::endl;
   }
 
-  std::shared_ptr<thread_pool> io_thread_pool = std::make_shared<thread_pool>();
+  auto io_thread_pool = make_unique<thread_pool>();
```

Comment:
@benson31 shouldn't this be std::make_unique

Reply:
Or is this our internal make_unique since we aren't c++14

Reply:
It's `lbann::make_unique`, which is `std::make_unique` if that's available.